cassandane/Cassandane/Mboxname.pm

```diff
@@ -161,20 +161,68 @@ sub from_external
     die "Bad external name \"$s\""
         if !defined $userid || $prefix ne 'user';
 
+    map { s/\./\^/g } @comps;
```

Comment:
nit: map in void is usually not needed, instead `s/// for @comps;`

Bigger question: Why assume here instead of checking config?

Reply:
(I do not particularly like this API fwiw, but I don't have a better one in mind yet.  I've used it enough to be annoyed by it, but not enough to know what I'd prefer.)

The next line passes these comps to `_set()`, which expects them to be in the internal format, where dots within mailbox name components are always represented by `^`, regardless of config.

I used this API both ways (i.e. with/without unixhierarchysep) while developing these improvements, and found this part wanted to be unconditional to be ergonomic.  Otherwise, when constructing an Mboxname object, you'd need to be mindful of the config, defeating the purpose of the module managing that detail for you.

Perhaps a better fix would be for `from_external` to take a namespace argument like `to_external` now does, and then die if the input format doesn't match match the format the declared namespace dictates.    But I didn't need that complexity, so I didn't add it.

This is mostly just a crude perl rewrite of the actual mangling that happens in mboxname.c.  So perhaps the even better rewrite would be exposing that so it can be used from perl.